### PR TITLE
Fix getting database config. Create default blueprint

### PR DESCRIPTION
**Pull request by asamofal**
Hi there, first of all thanks for the package! I've found a few issues that should be fixed...

### Database config issues
There are were a few issues with publishing the config file.
1) In `boot` method of `StatamicDatabaseServiceProvider` config is ready to be published to `config_path('statamic/statamic-database.php')`, but then in `register` method there's was a call just from the root `config('statamic-database')`. 
So I've changed the destination path to `config_path('statamic/database.php')` _(no sense to duplicate `statamic` word when we already in `statamic` config "namespace")_.

2) Publishing is an optional action. So in `register` method we should be ready for the case when config doesn't contain `'statamic.database'` key. Otherwise we get the error everytime: `Trying to access array offset on value of type null`, because `$config` variable is null.
To handle it I've added a call to `mergeConfigFrom` that applies the default configuration when the key doesn't exist ([doc](https://laravel.com/docs/8.x/packages#default-package-configuration)).

### Default Blueprint
Out of the box Statamic expects to have the `default` Blueprint _(`resources/blueprints/default.yml` file)_. 
As I've figured out, the scheme is like this:
The default Blueprint is used when user creates a new collection. On `Blueprints` page you may noticed a new Blueprint _(with the same name as collection)_, but in fact that Blueprint is not saved to the storage _(in file or record in DB)_. It's saved only when you edited it and click the "save" button .
So the default Blueprint is required for Statamic. If `blueprints` table is empty we can't create any new collections or even can't open the `Blueprints` page on the control panel.

I've created  `DefaultBlueprintSeeder` and run it automatically in `blueprints_table` migration. I know, migration is not the best place to do it, but otherwise it's necessary to force the users publish and run this seeder manually.
I've added a special exception for the case when default Blueprint is missing.

Also `DefaultBlueprintSeeder` is ready to be published. Users may publish it by the command:
```bash
php artisan vendor:publish --tag=statamic-database-seeders`
```
The seeder uses `updateOrInsert` DB command, so even after an automatic call from the migration the user will be able to publish it, customize and rerun to update the default Blueprint.

ps. I've moved `database` folder to `src/Database` in conformity with PSR-4 standard.

### CODE REVIEW

**Comment by Daynnnnn** [comment]
Awesome, these are 2 things I've noticed and had on my backlog - so thanks for sorting!

File: src/Blueprints/BlueprintRepository.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace Daynnnnn\StatamicDatabase\Blueprints;
 
 use Daynnnnn\StatamicDatabase\Blueprints\Traits\ExistsAsModel;
+use Daynnnnn\StatamicDatabase\Exceptions\DefaultBlueprintNotFoundException;
```
Comment (Daynnnnn) [inline src/Blueprints/BlueprintRepository.php:6]: I don't think this class is being defined anywhere?

**Review by Daynnnnn — CHANGES_REQUESTED** [review]
Had a look over and this is the only issue I can see, if you're okay to sort that when you get a chance I'm happy to merge. If not let me know and I should be able to sort :)

File: src/Blueprints/BlueprintRepository.php (hunk 1 above)
Comment (asamofal) [reply src/Blueprints/BlueprintRepository.php:6]: oh... yes, you're right, just forgot to commit the file. Fixed :)

**Comment by Daynnnnn** [comment]
Amazing, thank you!! 🙌

**Pull request merged**